docker-compose.stack.yml

```diff
@@ -0,0 +1,138 @@
+version: "3.7"
+
+x-custom:
+  shared_image: &shared_image
+    image: smirart/postgres-xl:kubernetes
+
+services:
+  pgpool:
+    image: smirart/pgpool:latest
+    ports:
+      - "5432:5432"
+    volumes:
+      - ./pgpool.conf:/etc/pgpool/pgpool.conf
+    restart: always
+    networks:
+      - db-b
+  db-gtm-1:
+    <<: *shared_image
+    environment:
+      - PG_HOST=0.0.0.0
+      - PG_NODE=gtm_1
+      - PG_PORT=6666
+    command: docker-cmd-gtm
+    entrypoint: docker-entrypoint-gtm
+    volumes:
+      - db-gtm-1-data:/var/lib/postgresql
+    networks:
+      - db-a
+    healthcheck:
+      test: ["CMD", "docker-healthcheck-gtm"]
+      interval: 30s
+      timeout: 10s
+      retries: 1
+  db-coord-1:
+    <<: *shared_image
+    environment:
+      - PG_GTM_HOST=db-gtm-1
+      - PG_GTM_PORT=6666
+      - PG_HOST=0.0.0.0
+      - PG_NODE=coord_1
+      - PG_PORT=5432
+      - WAIT_HOSTS=db-gtm-1:6666
+      - WAIT_BEFORE_HOSTS=20
+      - WAIT_HOSTS_TIMEOUT=60
+    command: sh -c "/wait && docker-cmd-coord"
+    entrypoint: docker-entrypoint-coord
+    volumes:
+      - db-coord-1-data:/var/lib/postgresql
+    networks:
+      - db-a
+      - db-b
+    healthcheck:
+      test: ["CMD", "docker-healthcheck-coord"]
+      interval: 30s
+      timeout: 10s
+      retries: 2
+  db-coord-2:
+    <<: *shared_image
+    environment:
+      - PG_GTM_HOST=db-gtm-1
+      - PG_GTM_PORT=6666
+      - PG_HOST=0.0.0.0
+      - PG_NODE=coord_2
```

Comment:
I'd be curious to hear more about ae8378f4e956022ddd7ed920965c7e062e88f3ec , if you would be so kind. It doesn't really matter what they're called—this is why it's even exposed as a variable—but it's interesting that it's different.

Reply:
I've reproduced this problem by `docker-compose.swarm.yml`

```bash
smirart@localhost postgres-xl-docker % docker stack deploy --orchestrator=kubernetes --namespace=default -c docker-compose.swarm.yml pxl_stack
top-level network "db_a" is ignored
top-level network "db_b" is ignored
service "db_data_2": network "db_a" is ignored
service "db_gtm_1": network "db_a" is ignored
service "db_coord_1": network "db_b" is ignored
service "db_coord_1": network "db_a" is ignored
service "db_coord_2": network "db_a" is ignored
service "db_coord_2": network "db_b" is ignored
service "db_data_1": network "db_a" is ignored
Stack.compose.docker.com "pxl_stack" is invalid: [spec.stack.services[0].name: Invalid value: "db_data_2": not a valid service name in Kubernetes: a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), spec.stack.services[0].volumes[0]: Invalid value: "db_data_2": not a valid volume name in Kubernetes: a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), spec.stack.services[1].name: Invalid value: "db_gtm_1": not a valid service name in Kubernetes: a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), spec.stack.services[1].volumes[0]: Invalid value: "db_gtm_1": not a valid volume name in Kubernetes: a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), spec.stack.services[2].name: Invalid value: "db_coord_1": not a valid service name in Kubernetes: a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), spec.stack.services[2].volumes[0]: Invalid value: "db_coord_1": not a valid volume name in Kubernetes: a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), spec.stack.services[3].name: Invalid value: "db_coord_2": not a valid service name in Kubernetes: a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), spec.stack.services[3].volumes[0]: Invalid value: "db_coord_2": not a valid volume name in Kubernetes: a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), spec.stack.services[4].name: Invalid value: "db_data_1": not a valid service name in Kubernetes: a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), spec.stack.services[4].volumes[0]: Invalid value: "db_data_1": not a valid volume name in Kubernetes: a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')]
```

Errors:

spec.stack.services[0].name: Invalid value: "db_data_2": not a valid service name in Kubernetes: a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'),

spec.stack.services[0].volumes[0]: Invalid value: "db_data_2": not a valid volume name in Kubernetes: a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')

Reply:
Thanks!